src/Core/StringLocalizer.cs

```diff
@@ -1,31 +1,42 @@
-﻿using Microsoft.Extensions.Localization;
-using System;
+﻿using System;
 using System.Collections.Generic;
 using System.Globalization;
+using System.Linq;
+using Microsoft.Extensions.Localization;
 
 namespace Core
 {
-    public class StringLocalizer : IStringLocalizer
-    {
-        readonly Func<IReadOnlyDictionary<string, IReadOnlyDictionary<string, string>>> modelProvider;
+	public class StringLocalizer : IStringLocalizer
+	{
+		private readonly Func<IReadOnlyDictionary<string, IReadOnlyDictionary<string, string>>> _modelProvider;
+
+        private CultureInfo _culture;
+        private CultureInfo Culture => _culture ?? CultureInfo.CurrentUICulture;
 
         public StringLocalizer(Func<IReadOnlyDictionary<string, IReadOnlyDictionary<string, string>>> modelProvider)
```

Comment:
Since StringLocalizer gets the culture in the constructor, you can lookup the dictionary for the culture and store that in a field. Then all methods can use that field and can skip one dictionary lookup.

Reply:
Of course that could be a problem if StringLocalizer is a singleton per culture. Not sure about the lifetime of those...

Reply:
I don't really think we should do that. I just created that second constructor for convenience, expecting that we would need it in the future. 

In an effort to give control of fetching/caching of the strings to another object, the string localizer always needs to look up the string using the _modelProvider for the relevant culture. Thus, it is the _modelProvider's responsibility to be efficient about it.